tests/integration/tests/autocomplete.js

```diff
@@ -1,59 +1,116 @@
 import {initBrowser, wait} from '../tools'
+import AutocompleteCucumberise from "../cucumberise/autocomplete";
 const configBuilder = require('@qwant/nconf-builder')
 const config = configBuilder.get()
 const APP_URL = `http://localhost:${config.PORT}`
 
 let browser
 let page
+let autocompleteHelper
 const mockAutocomplete = require('../../__data__/autocomplete')
 
 beforeAll(async () => {
   let browserPage = await initBrowser()
   page = browserPage.page
   browser = browserPage.browser
-  await page.setRequestInterception(true)
-  page.on('request', interceptedRequest => {
-    if(interceptedRequest.url().match(/autocomplete/)) {
-      interceptedRequest.headers['Access-Control-Allow-Origin'] = '*'
-      interceptedRequest.respond({body : JSON.stringify(mockAutocomplete), headers  : interceptedRequest.headers})
-    } else {
-      interceptedRequest.continue()
-    }
-  })
+  autocompleteHelper = new AutocompleteCucumberise(page)
+  await autocompleteHelper.prepareResponse()
 })
 
-test('clear button',async () => {
-  expect.assertions(3)
+test('search and clear', async () => {
+  expect.assertions(4)
+  autocompleteHelper.addPreparedResponse(mockAutocomplete, /autocomplete/)
```

Comment:
if it's a "hardcoded" response, shouldn't it be set (instead of added) ?

Reply:
maybe to early but it would support multiple query / response handling.